src/field_theory/minimal_polynomial.lean

```diff
@@ -90,7 +97,35 @@ begin
   simpa only [eq_one, alg_hom.map_one, one_ne_zero] using aeval hx
 end
 
-/--A minimal polynomial is not a unit.-/
+/-- If `L/K` is a ring extension, and `x` is an element of `L` in the image of `K`,
+then the minimal polynomial of `x` is `X - C x`. -/
+lemma algebra_map_inj [nontrivial α] (a : α) (hf : function.injective (algebra_map α β))
+  (ha : is_integral α (algebra_map α β a)) : minimal_polynomial ha = X - C a :=
```

Comment:
I am a bit confused by this. Isn't `is_integral α (algebra_map α β a)` always true?

Reply:
Yes, that's of course always true. What I did was to copy the statement of `algebra_map` (the exact same thing for fields) without too much thinking about it. I guess the reason why it is formulated this way was that usually when you apply it you have an integral element that in same specific case happens to be in the base ring, so you already have the needed assumption.

In any case I am writing a minilemma to avoid it.

Reply:
Ah, it is already there! `is_integral_algebra_map`